cmd/sonobuoy/app/gen.go

```diff
@@ -65,14 +69,46 @@ func (g *genFlags) Config() (*client.GenConfig, error) {
 		return nil, errors.Wrap(err, "could not retrieve E2E config")
 	}
 
+	kubeclient, kubeError := maybeGetClient(&g.kubecfg)
+
+	rbacEnabled, err := genflags.rbacMode.Enabled(kubeclient)
+	if err != nil {
+		if errors.Cause(err) == ErrRBACNoClient {
```

Comment:
This type of error checking is a bit fragile and also requires errors to be exported for testing. An alternative approach to consider is defining a behavior on the custom error and then type checking to see if the returned error implements that behavior:

https://golang.org/src/os/error.go#L22

This makes it easy to create an error that implements that behavior without having to do a string comparison.